src/Microsoft.Identity.Client/Internal/ClientCredentialWrapper.cs

```diff
@@ -33,6 +33,9 @@ public ClientCredentialWrapper(ClientAssertionCertificateWrapper certificate)
         internal long ValidTo { get; set; }
         internal bool ContainsX5C { get; set; }
         internal string Audience { get; set; }
+        internal string Secret { get; private set; }
+        internal ClientAssertion UserProvidedClientAssertion { get; set; }
```

Comment:
We now have 3 properties named Assertion - which ones does what? Consider adding comments to all of these properties.